google-cloud-firestore/src/main/java/com/google/cloud/firestore/TransactionOptions.java

```diff
@@ -16,65 +16,133 @@
+/**
+ * Options specifying the behavior of Firestore Transactions.
+ *
+ * <p>A transaction in Firestore can be either read-write or read-only.
+ *
+ * <p>The default set of options is a read-write transaction with a maximum number of 5 attempts.
+ * This attempt count can be customized via the {@link
+ * ReadWriteOptionsBuilder#setNumberOfAttempts(int)} method. A new instance of a builder can be
+ * created by calling {@link #createReadWriteOptionsBuilder()}.
+ *
+ * <p>A read-only transaction can be configured via the {@link ReadOnlyOptionsBuilder} class. A new
+ * instance can be created by calling {@link #createReadOnlyOptionsBuilder()}.
+ *
+ * @see com.google.firestore.v1.TransactionOptions
+ */
 public final class TransactionOptions {
 
+  private static final TransactionOptions DEFAULT_READ_WRITE_TRANSACTION_OPTIONS =
+      createReadWriteOptionsBuilder().build();
+
   private static final int DEFAULT_NUM_ATTEMPTS = 5;
 
-  private final int numberOfAttempts;
   private final Executor executor;
+  private final TransactionOptionsType type;
+  private final int numberOfAttempts;
+  @Nullable private final Timestamp readTime;
 
-  TransactionOptions(int maxAttempts, Executor executor) {
-    this.numberOfAttempts = maxAttempts;
+  TransactionOptions(
+      Executor executor,
+      TransactionOptionsType type,
+      int numberOfAttempts,
+      @Nullable Timestamp readTime) {
     this.executor = executor;
+    this.type = type;
+    this.numberOfAttempts = numberOfAttempts;
+    this.readTime = readTime;
   }
 
+  /**
+   * Returns the maximum number of times a transaction will be attempted before resulting in an
+   * error.
+   *
+   * @return The max number of attempts to try and commit the transaction.
+   */
   public int getNumberOfAttempts() {
     return numberOfAttempts;
   }
 
+  /** @return Executor to be used to run user callbacks on */
   @Nullable
   public Executor getExecutor() {
     return executor;
   }
 
   /**
-   * Create a default set of options suitable for most use cases. Transactions will be attempted 5
-   * times.
+   * A type flag indicating the type of transaction represented.
+   *
+   * @return The type of transaction this represents. Either read-only or read-write.
+   */
+  @Nonnull
+  public TransactionOptionsType getType() {
+    return type;
+  }
+
+  /**
+   * A {@link Timestamp} specifying the time documents are to be read at. If null, the server will
+   * read documents at the most up to date available. If nonnull, the specified {@code Timestamp}
+   * may not be more than 60 seconds in the past (evaluated when the request is processed by the
+   * server).
+   *
+   * @return The specific time to read documents at. A null value means read most up to date data.
+   */
+  @Nullable
+  public Timestamp getReadTime() {
+    if (TransactionOptionsType.READ_ONLY.equals(type)) {
+      return readTime;
+    } else {
+      return null;
+    }
```

Comment:
This if statement is technically not needed, but it might help others understand how this works.

Reply:
Yeah, I kept it in here for clarity sake. I'll add a comment next to it stating as much.